### PR TITLE
Add documentation for the sqlite backend options

### PR DESCRIPTION
**Pull request by espadolini**
This PR adds some description of parameters of the `sqlite` backend that were previously undocumented, and adds some advice on what to do about poor performance when using SQLite as a backend (the infamous `SLOW TRANSACTION` log spam) after the changes in #11387.

### CODE REVIEW

File: docs/pages/setup/reference/backends.mdx [hunk 1]
```diff
@@ -684,3 +684,49 @@ $ psql -d postgres
...
+The SQLite backend is used when no `type` is specified in the storage section in
+the Teleport configuration file, or when `type` is set to `sqlite` or `dir`.
+It's not designed for high throughput and it's not capable of serving the needs
+of Teleport's High Availability configurations; as such, it's only recommended
+for deployments of moderate size.
+
+A sign that the cluster has outgrown the capabilities of the SQLite backend in a
+local directory is the presence of warning messages in the auth server's logs
+that say `SLOW TRANSACTION`. As a stopgap measure until it's possible to migrate
+the cluster to use a HA-capable backend, the SQLite backend can be configured to
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:699]:
```suggestion
the cluster to use a HA-capable backend, you can configure the SQLite backend to
```
to use the active voice

File: docs/pages/setup/reference/backends.mdx [hunk 2]
```diff
@@ -684,3 +684,49 @@ $ psql -d postgres
...
+
+The SQLite backend is used when no `type` is specified in the storage section in
+the Teleport configuration file, or when `type` is set to `sqlite` or `dir`.
+It's not designed for high throughput and it's not capable of serving the needs
+of Teleport's High Availability configurations; as such, it's only recommended
+for deployments of moderate size.
+
+A sign that the cluster has outgrown the capabilities of the SQLite backend in a
+local directory is the presence of warning messages in the auth server's logs
+that say `SLOW TRANSACTION`. As a stopgap measure until it's possible to migrate
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:698]:
```suggestion
that say `SLOW TRANSACTION`. 

As a stopgap measure until it's possible to migrate
```
The paragraph is a little long

File: docs/pages/setup/reference/backends.mdx [hunk 3]
```diff
@@ -684,3 +684,49 @@ $ psql -d postgres
...
+the cluster to use a HA-capable backend, the SQLite backend can be configured to
+reduce the amount of disk synchronization, in exchange for less resilience
+against system crashes or power loss. For an explanation on what the options
+mean, see [the official SQLite
+docs](https://www.sqlite.org/pragma.html#pragma_synchronous). No matter the
+configuration, we recommend you take regular backups of your cluster state.
+
+```yaml
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:706]: I think we need to connect these two code snippets to the preceding paragraph more explicitly.

File: docs/pages/setup/reference/backends.mdx [hunk 4]
```diff
@@ -684,3 +684,49 @@ $ psql -d postgres
...
 postgres=# GRANT ALL PRIVILEGES ON DATABASE teleport TO dbuser;
 ```
+
+## SQLite
+
+The SQLite backend is used when no `type` is specified in the storage section in
+the Teleport configuration file, or when `type` is set to `sqlite` or `dir`.
+It's not designed for high throughput and it's not capable of serving the needs
+of Teleport's High Availability configurations; as such, it's only recommended
+for deployments of moderate size.
```
[the remark this answers is not in the document]
Comment (espadolini) [reply docs/pages/setup/reference/backends.mdx:694]: I started adding something along the lines of

```
The load on the backend depends on both how many resources are connected to the
cluster and how active said cluster is, and how well SQLite can cope with the load depends heavily on the IO resources available to the machine running the Auth Service, so we can't give any numbers here, but a sign that [...]
```

but that feels even worse than saying nothing, somehow.

File: docs/pages/setup/reference/backends.mdx [hunk 5]
```diff
@@ -684,3 +684,52 @@ $ psql -d postgres
...
+## SQLite
+
+The Auth Service uses the SQLite backend when no `type` is specified in the
+storage section in the Teleport configuration file, or when `type` is set to
+`sqlite` or `dir`. It's not designed for high throughput and it's not capable of
+serving the needs of Teleport's High Availability configurations; as such, it's
+only recommended for deployments of moderate size.
+
+A sign that the cluster has outgrown the capabilities of the SQLite backend in a
+local directory is the presence of warning messages in the Auth Service's logs
```
Comment (zmb3) [inline docs/pages/setup/reference/backends.mdx:697]:
```suggestion
A sign that the cluster has outgrown the capabilities of the SQLite backend is the presence of warning messages in the Auth Service's logs
```

**Review by zmb3 — APPROVED** [review]

File: docs/pages/setup/reference/backends.mdx [hunk 6]
```diff
@@ -684,3 +684,52 @@ $ psql -d postgres
...
 postgres=# GRANT ALL PRIVILEGES ON DATABASE teleport TO dbuser;
 ```
+
+## SQLite
+
+The Auth Service uses the SQLite backend when no `type` is specified in the
+storage section in the Teleport configuration file, or when `type` is set to
+`sqlite` or `dir`. It's not designed for high throughput and it's not capable of
+serving the needs of Teleport's High Availability configurations; as such, it's
+only recommended for deployments of moderate size.
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:694]: To make the scaling recommendations more specific and actionable, what would you think of:

- Removing "; as such, it's
only recommended for deployments of moderate size"
- In the next paragraph, add a sentence similar to, "If you are planning to use SQLite as your backend, scale your cluster slowly and monitor the number of `SLOW TRANSACTION` logs".

**Review by ptgott — APPROVED** [review]

**Pull request merged**